### PR TITLE
Add Single page apps in depth by Mixu

### PR DESCRIPTION
**Pull request by uuykay**
## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource(s) | Remove Resource(s) | Add info | Improve Repo

## For resources
Javascript

### Description 
Add Single page apps in depth by Mixu

### Why is this valuable (or not)?
Explains SPA architecture.

### How do we know it's really free?
Mentioned by author and is on github.

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

**Comment by SethiPandi** [comment]
Hey, and welcome to free programming books! :wave:

It looks like you've let your code editor auto-format the document and break most of the repositories conventions. It's always worth giving the changes files a peek before submitting the PR! Could you please fix the formatting?

**Comment by uuykay** [comment]
> Hey, and welcome to free programming books! 👋
> 
> It looks like you've let your code editor auto-format the document and break most of the repositories conventions. It's always worth giving the changes files a peek before submitting the PR! Could you please fix the formatting?

Sorry! I tried disabling Prettier in the workspace, but ended up having to just open it in Notepad to edit the original file.

File: free-programming-books.md [hunk 1]
```diff
@@ -1698,6 +1698,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Professor Frisby’s Mostly Adequate Guide to Functional Programming](https://mostly-adequate.gitbooks.io/mostly-adequate-guide/content/) - Brian Lonsdorf
 * [Programming JavaScript Applications](https://www.oreilly.com/library/view/programming-javascript-applications/9781491950289/)
 * [Robust Client-Side JavaScript](https://molily.de/robust-javascript/) - Matthias Schäfer (HTML, EPUB)
+* [Single page apps in depth](http://singlepageappbook.com/) - Mixu (HTML)
```
Comment (SethiPandi) [inline free-programming-books.md:1701]: For root URLs we prefer not to have a trailing slash, could you remove it, please?

```suggestion
* [Single page apps in depth](http://singlepageappbook.com) - Mixu (HTML)
```

**Review by SethiPandi — CHANGES_REQUESTED** [review]
You can get more guidance on the formatting and conventions in this repository by checking out the [contribution guide](https://www.github.com/EbookFoundation/free-programming-books/tree/master/CONTRIBUTING.md).

You can also click on the failed checks to get more information on any check failures. You'll be able to see exactly what was wrong and how to fix it!

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**